### PR TITLE
Anomaly #47389 : Checking before changing status : Stock Management

### PR DESCRIPTION
**Pull request by alo-axelor**

### CODE REVIEW

File: axelor-supplychain/src/main/resources/views/MrpForecast.xml [hunk 1]
```diff
@@ -76,12 +76,24 @@
...
+  <action-group name="action-mrp-forecast-group-confirm-button">
+    <action name="save"/>
+    <action name="action-mrp-forecast-method-confirm"/>
+    <action name="save"/>
+  </action-group>
+
+  <action-group name="action-mrp-forecast-group-cancel-button">
+    <action name="save"/>
+    <action name="action-mrp-forecast-method-cancel"/>
+    <action name="save"/>
```
Comment (ale-axelor) [inline axelor-supplychain/src/main/resources/views/MrpForecast.xml:96]: Please remove the last save

File: axelor-supplychain/src/main/resources/views/MrpForecast.xml [hunk 2]
```diff
@@ -76,12 +76,24 @@
...
-    <attribute name="value" for="statusSelect" expr="eval: 3"/>
-  </action-attrs>
+  <action-method name="action-mrp-forecast-method-cancel">
+    <call class="com.axelor.apps.production.web.MrpForecastController" method="cancel"/>
+  </action-method>
+
+  <action-group name="action-mrp-forecast-group-confirm-button">
+    <action name="save"/>
+    <action name="action-mrp-forecast-method-confirm"/>
+    <action name="save"/>
```
Comment (ale-axelor) [inline axelor-supplychain/src/main/resources/views/MrpForecast.xml:90]: Please remove the last save, now that we use an action method fetching from db it is not necessary anymore.

File: axelor-stock/src/main/java/com/axelor/apps/stock/service/StockMoveServiceImpl.java [hunk 3]
```diff
@@ -373,6 +380,13 @@ public String realize(StockMove stockMove) throws AxelorException {
   @Transactional(rollbackOn = {Exception.class})
   public String realize(StockMove stockMove, boolean checkOngoingInventoryFlag)
```
Comment (ale-axelor) [inline axelor-stock/src/main/java/com/axelor/apps/stock/service/StockMoveServiceImpl.java:381]: Please remove the previously added check for stock move realization in StockMoveController

Comment (alo-axelor) [reply axelor-stock/src/main/java/com/axelor/apps/stock/service/StockMoveServiceImpl.java:381]: I remove the check in StockMoveController in the first commit of the PR.

File: axelor-supplychain/src/main/resources/views/MrpForecast.xml (hunk 1 above)
Comment (alo-axelor) [reply axelor-supplychain/src/main/resources/views/MrpForecast.xml]: Done

File: axelor-supplychain/src/main/resources/views/MrpForecast.xml (hunk 2 above)
Comment (alo-axelor) [reply axelor-supplychain/src/main/resources/views/MrpForecast.xml]: Done

File: axelor-production/src/main/java/com/axelor/apps/production/service/MrpForecastService.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package com.axelor.apps.production.service;
+
+import com.axelor.apps.supplychain.db.MrpForecast;
+import com.axelor.exception.AxelorException;
+
+public interface MrpForecastService {
```
Comment (kra-axelor) [inline axelor-production/src/main/java/com/axelor/apps/production/service/MrpForecastService.java:6]: Please add javadoc on the methods

Comment (alo-axelor) [reply axelor-production/src/main/java/com/axelor/apps/production/service/MrpForecastService.java:6]: Done

**Review by kra-axelor — APPROVED** [review]

**Pull request merged**